src/ray/raylet/worker_pool.cc

```diff
@@ -651,9 +688,13 @@ void WorkerPool::OnWorkerStarted(const std::shared_ptr<WorkerInterface> &worker)
 
   auto it = state.starting_worker_processes.find(worker_startup_token);
   if (it != state.starting_worker_processes.end()) {
+    runtime_env_manager_.AddURIReference(worker->WorkerId().Hex(),
```

Comment:
Isn't this having a leak when `num_starting_workers ` > 1? For example, 

`OnWorkerStarted ` will be called per each starting workers, but `RemoveURIReference ` will be called only once when num_starting_workers == 0?

Reply:
We remove reference in `DisconnectWorker` which is triggered by echo worker instance.

Reply:
Sorry what does `echo worker instance.` mean?

Reply:
Sorry, echo -> each